testsuite/python/collision_detection_interface.py

```diff
@@ -53,35 +53,22 @@ class CollisionDetection(ut.TestCase):
             "part_type_to_be_glued": 2, "part_type_after_glueing": 3
         },
     }
+    classes = {"bind_centers": espressomd.collision_detection.BindCenters,
+               "bind_at_point_of_collision": espressomd.collision_detection.BindAtPointOfCollision,
+               "glue_to_surface": espressomd.collision_detection.GlueToSurface}
 
     def tearDown(self):
-        self.system.collision_detection.set_params(mode="off")
+        self.system.collision_detection.protocol = None
 
     def test_00_interface_and_defaults(self):
-        # Is it off by default
-        self.assertEqual(self.system.collision_detection.mode, "off")
-
-        # Make sure params cannot be set individually
-        with self.assertRaises(Exception):
-            self.system.collision_detection.mode = "bind_centers"
-
-        # Verify exception throwing for unknown collision modes
-        with self.assertRaisesRegex(ValueError, "Unknown collision mode 'unknown'"):
-            self.system.collision_detection.set_params(mode="unknown")
-        with self.assertRaisesRegex(ValueError, "Collision mode must be specified via the 'mode' argument"):
-            self.system.collision_detection.set_params()
-
+        self.assertIsNone(self.system.collision_detection.protocol, None)
```

Comment:
I think this is confusing, we should either provide a useful error message or not state the default value explicitly.
```suggestion
        self.assertIsNone(self.system.collision_detection.protocol)
```